src/read_write/cube.rs

```diff
@@ -163,71 +164,104 @@ pub struct CubeNodeWriter {
-                    .unwrap();
-                self.xyz_writer
-                    .write_u8(fixpoint_encode(p.position.y, min.y, edge_length))
-                    .unwrap();
-                self.xyz_writer
-                    .write_u8(fixpoint_encode(p.position.z, min.z, edge_length))
-                    .unwrap();
+                for position in &p.position {
+                    vec3_fixpoint_encode::<u8>(position, min, edge_length)
+                        .write_le(&mut self.xyz_writer)?;
+                }
             }
             PositionEncoding::Uint16 => {
-                self.xyz_writer
-                    .write_u16::<LittleEndian>(fixpoint_encode(p.position.x, min.x, edge_length))
-                    .unwrap();
-                self.xyz_writer
-                    .write_u16::<LittleEndian>(fixpoint_encode(p.position.y, min.y, edge_length))
-                    .unwrap();
-                self.xyz_writer
-                    .write_u16::<LittleEndian>(fixpoint_encode(p.position.z, min.z, edge_length))
-                    .unwrap();
+                for position in &p.position {
+                    vec3_fixpoint_encode::<u16>(position, min, edge_length)
+                        .write_le(&mut self.xyz_writer)?;
+                }
             }
             PositionEncoding::Float32 => {
-                self.xyz_writer
-                    .write_f32::<LittleEndian>(encode(p.position.x, min.x, edge_length))
-                    .unwrap();
-                self.xyz_writer
-                    .write_f32::<LittleEndian>(encode(p.position.y, min.y, edge_length))
-                    .unwrap();
-                self.xyz_writer
-                    .write_f32::<LittleEndian>(encode(p.position.z, min.z, edge_length))
-                    .unwrap();
+                for position in &p.position {
+                    vec3_encode::<f32>(position, min, edge_length)
+                        .write_le(&mut self.xyz_writer)?;
+                }
             }
             PositionEncoding::Float64 => {
-                self.xyz_writer
-                    .write_f64::<LittleEndian>(encode(p.position.x, min.x, edge_length))
-                    .unwrap();
-                self.xyz_writer
-                    .write_f64::<LittleEndian>(encode(p.position.y, min.y, edge_length))
-                    .unwrap();
-                self.xyz_writer
-                    .write_f64::<LittleEndian>(encode(p.position.z, min.z, edge_length))
-                    .unwrap();
+                for position in &p.position {
+                    vec3_encode::<f64>(position, min, edge_length)
+                        .write_le(&mut self.xyz_writer)?;
+                }
             }
         }
 
-        p.color.write_le(&mut self.attribute_writers[0]);
-        if let Some(i) = p.intensity {
-            if self.attribute_writers.len() < 2 {
+        if self.point_count == 0 {
```

Comment:
Does PointData always contain > 0 points?

Reply:
Theoretically it can contain anything from 0 to NUM_POINTS_PER_BATCH...

Reply:
Maybe this should test `self.attribute_writers.len()` instead?

Reply:
Unfortunately this would repeat the check for each iteration, if we don't write any attributes.

Reply:
This would be possible, but would be slightly slower, as we needed two check when writing no attributes at all.

Reply:
Hm, but since we almost always write attributes, it is nicer, since we can get rid of `point_count`.

Reply:
Done, with `self.attribute_writers.is_empty()`

Reply:
:+1: